### PR TITLE
MRG: Fix examples

### PR DESCRIPTION
**Pull request by Eric89GXL**
This should fix the Neo problem in a backward- and future-compatible way.

Also updates a bunch of `filter` calls to avoid deprecationwarnings.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
I've gone through and looked at the examples. They all look pretty similar, except that the covariance one gets a little bit better:

https://mne-tools.github.io/dev/auto_examples/visualization/plot_evoked_whitening.html
https://4323-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/auto_examples/visualization/plot_evoked_whitening.html

This one changes but I can't tell if it's better or worse:

https://mne-tools.github.io/dev/auto_examples/decoding/plot_decoding_unsupervised_spatial_filter.html
https://4323-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/auto_examples/decoding/plot_decoding_unsupervised_spatial_filter.html

**Comment by agramfort** [comment]
LGTM

@kingjr let us know if PCA/ICA example still looks good for you.

File: tutorials/plot_artifacts_correction_ica.py [hunk 1]
```diff
@@ -30,7 +30,8 @@
 raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
 
 raw = mne.io.read_raw_fif(raw_fname, preload=True)
-raw.filter(1, 40, n_jobs=2)  # 1Hz high pass is often helpful for fitting ICA
+# 1Hz high pass is often helpful for fitting ICA
+raw.filter(1, 40, n_jobs=2, fir_design='firwin')
```
Comment (kingjr) [inline tutorials/plot_artifacts_correction_ica.py]: maybe float freqs

File: tutorials/plot_modifying_data_inplace.py [hunk 2]
```diff
@@ -36,8 +36,7 @@
 _ = ax.plot(raw._data[0])
 for fband in filt_bands:
```
Comment (kingjr) [inline tutorials/plot_modifying_data_inplace.py]: explicit fmin, fmax?

**Comment by Eric89GXL** [comment]
Thanks @kingjr. I'll merge so that CircleCI can build properly on `master`

**Pull request merged**